packages/rocketchat-lib/server/lib/processDirectEmail.js

```diff
@@ -81,10 +81,27 @@ RocketChat.processDirectEmail = function(email) {
 		}
 
 		if ((room.muted || []).includes(user.username)) {
-			// room is muted
+			// user is muted
 			return false;
 		}
 
+		if (room.ro === true) {
+			const userOwner = RocketChat.models.Roles.findOne({
```

Comment:
I don't think we should test room's owner. Since `post-readonly` is given by default to _owners_ and _moderators_ I think we can test only the permission.